src/nspawn/nspawn-network.c

```diff
@@ -210,6 +236,40 @@ static void shorten_ifname(char *ifname) {
         /* Log the incident to make it more discoverable */
         log_warning("Network interface name '%s' has been changed to '%s' to fit length constraints.", ifname, new_ifname);
 
+        strcpy(ifname, new_ifname);
+        return 1;
+}
+
+static void shorten_alternative_ifname(char *ifname) {
+        char new_ifname[ALTIFNAMSIZ];
+        uint64_t h;
+
+        assert(ifname);
+
+        if (strlen(ifname) < ALTIFNAMSIZ) /* Name is short enough */
+                return;
+
+        /* Calculate 64bit hash value */
+        h = siphash24(ifname, strlen(ifname), SHORTEN_IFNAME_HASH_KEY.bytes);
+
+        /* Set the final four bytes (i.e. 32bit) to the lower 24bit of the hash, encoded in url-safe base64 */
+        memcpy(new_ifname, ifname, ALTIFNAMSIZ - 12);
```

Comment:
hmm, the comment doesn't match the code here....

But i wonder if we should really bother with the hashing here....  i mean, we hash for the regular name, since we have to come up with a name of some form, and that's a way to get there. However, alternative names are optional, hence I'd just not do them if we can't make the name fit.

Besides that we insist that container names are valid hostnames and for those we insist on a size limit of HOST_NAME_MAX which is 64. ALTIFNAMSIZ is 128 however. Which means the hostnames should always fit fine in the alternative names anyway...

Hence, my suggestion: if the name passed here doesn't fit the size limit, just don't set an alternative name...